src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -351,3 +367,419 @@ deriving stock instance Read SizeRatio
 deriving stock instance Read MergePolicy
 deriving stock instance Read BloomFilterAlloc
 deriving stock instance Read FencePointerIndex
+
+{-------------------------------------------------------------------------------
+  Encoding and decoding
+-------------------------------------------------------------------------------}
+
+class Encode a where
+  encode :: a -> Encoding
+
+class Decode a where
+  decode :: Decoder s a
+
+class DecodeVersioned a where
+  decodeVersioned :: SnapshotVersion -> Decoder s a
```

Comment:
So the convention here is that things like `SnapshotVersion` and `Versioned` are in class `Decode`, and everything else is in `DecodeVersioned` right? Would be a helpful reader's guide to add this as a comment.